pages/en/zkapps/simple-anonymous-message-board-tutorial.mdx

```diff
@@ -0,0 +1,345 @@
+
+```ts
+// Computer new messageHistoryHash
+const oldHash = this.messageHistoryHash.get();
+const newHash = Poseidon.hash([message, oldHash]);
+
+// Update on-chain messageHistoryHash
+this.messageHistoryHash.set(newHash);
+```
+
+That’s it! Save the file, and let’s make sure everything compiles.
+
+```sh
+npm run build
+```
+
+If everything is correct, you should see a new `./build` directory. This is where the compiled version of your project lives that you can import into a user interface.
+
+## Integrating with a User Interface
+
+One of the benefits of writing smart contracts in TypeScript is that they are usually straightforward to <DocLink copy="integrate with UI components" url="/zkapps/how-to-write-a-zkapp" />.
+
+We won’t make you write a UI for this tutorial, but we have created a simple command-line interface so you can interact with your smart contract as a user, get a sense of how it works, and experiment with what you have just built. You can clone it from [this](https://github.com/) repository.
+
+```sh
+git clone https://placeholder.com
+cd ./placeholder
+```
+
+Now install your smart contract:
+
+```sh
+npm install ../zk-message
+```
+
+And start the front end:
+
+```sh
+npm run start
+
+Who are you? (Bob, SuperBob, MegaBob, Jack): Bob
+
+What would you like to say? (should only work if you are one of the Bobs): Snarky is a nice sharky.
+
+Message history:
+Snarky is a nice sharky.
+
+Who are you? (Bob, SuperBob, MegaBob, Jack): Jack
+
+What would you like to say? (should only work if you are one of the Bobs): Hello World!
+
+** GNARLY ERROR **
+```
+
+## Wrapping up
+
+Hopefully you enjoyed this tutorial, and it gave you a sense of what's possible with SnarkyJS. The messaging protocol we built is quite simple but also very powerful. This basic pattern could be used to create a whistleblower system, an anonymous NFT project, or even anonymous DAO voting. SnarkyJS makes it easy for developers to build things that don’t intuitively seem possible, and this is really the point. Zero-knowledge proofs open the door to an entirely different way of thinking about the internet, and we are so excited to see what people like you will build. Make sure to join the #zkapps-developers channel on [our Discord](https://bit.ly/MinaDiscord), and apply for the [zkApp Builders Program](https://share.hsforms.com/1KUeCPxJzQC6PIZ-LLdZe1g4xuul) if you are interested in building something more complex with the help of our team at O(1) Labs. More tutorials are on the way ([suggest ideas](https://docs.google.com/forms/d/e/1FAIpQLSctM8rq5B7teYhGl8hi2mk8NoSs6YiCDqjOnUfzpAAymU6Uug/viewform?usp=sf_link)), but these are some logical next steps if you are interested in extending this project.
```

Comment:
Would suggest adding a paragraph before `Zero-knowledge proofs open the door...`

We could link up `#zkapps-developers` directly to https://discord.com/channels/484437221055922177/915745847692636181 

`with the guidance of our team at O(1) Labs` (changed help to guidance)